File: Selenium/Activities/Activity_9.java
package Activities;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity_9 {
    public static void main(String[] args) {
        WebDriver driver = new ChromeDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(20));
        driver.get("https://v1.training-support.net/selenium/ajax");
        System.out.println("The tile of page is: " + driver.getTitle());
        driver.findElement(By.className("violet")).click();
        wait.until(ExpectedConditions.presenceOfElementLocated(By.xpath("//div[@id='ajax-content']/h1")));
        String msg1 = driver.findElement(By.xpath("//div[@id='ajax-content']/h1")).getText();
        System.out.println(msg1);
        wait.until(ExpectedConditions.textToBePresentInElementLocated(By.xpath("//div[@id='ajax-content']/h3"),"I'm late!"));
        String msg2 = driver.findElement(By.xpath("//div[@id='ajax-content']/h3")).getText();
        System.out.println(msg2);
        driver.quit();


    }
}

File: Selenium/Activities/Activity_9.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.wait import WebDriverWait

with webdriver.Chrome() as driver:
    driver.get("https://v1.training-support.net/selenium/ajax")
    print("The page title is: ", driver.title)
    mywait = WebDriverWait(driver, 20)
    driver.find_element(By.CLASS_NAME, "violet").click()
    ele1 = mywait.until(EC.presence_of_element_located((By.XPATH, "//div[@id='ajax-content']/h1")))
    print(ele1.text)
    mywait.until(EC.text_to_be_present_in_element((By.XPATH, "//div[@id='ajax-content']/h3"), "I'm late!"))
    ele2 = driver.find_element(By.XPATH, "//div[@id='ajax-content']/h3")
    print(ele2.text)

    driver.quit()


